processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,7 +88,20 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
-	return d.rb.Emit(), conventions.SchemaURL, nil
+	if d.ra.K8sClusterName.Enabled {
+		var clusterName string
+		clusterName, err = d.utils.getClusterName(ctx)
+		if err != nil {
+			// We clear out the error here so that the correctly detected resources
+			// can still be used by the processor.
+			d.logger.Warn("Unable to get EKS cluster name", zap.Error(err))
+			err = nil
+		} else {
+			d.rb.SetK8sClusterName(clusterName)
+		}
```

Comment:
Is this going to log each time a datapoint is processed or only once on startup? If it's each time then we should consider making this debug and not warn level. 

Also, manually setting the error here feels like an anti pattern. If the error is only used for a logging statement, and never returned to the caller, would it make more sense to not return an error at all from `getClusterName`? A logging statement could just be added inside that function.

Reply:
This is only logged once at collector startup.

Example logs: (collector is running with filelog receiver and debug exporter)
```
$ kubectl -n crobert-test logs otel-collector-7985f54766-dtdbf
2023-11-10T16:55:22.020Z	info	service@v0.88.1-0.20231026220224-6405e152a2d9/telemetry.go:84	Setting up own telemetry...
2023-11-10T16:55:22.020Z	info	service@v0.88.1-0.20231026220224-6405e152a2d9/telemetry.go:201	Serving Prometheus metrics	{"address": ":8888", "level": "Basic"}
2023-11-10T16:55:22.020Z	info	exporter@v0.88.1-0.20231026220224-6405e152a2d9/exporter.go:275	Development component. May change in the future.	{"kind": "exporter", "data_type": "logs", "name": "debug"}
2023-11-10T16:55:22.110Z	info	service@v0.88.1-0.20231026220224-6405e152a2d9/service.go:143	Starting otelcontribcol...	{"Version": "0.88.0-dev", "NumCPU": 16}
2023-11-10T16:55:22.110Z	info	extensions/extensions.go:33	Starting extensions...
2023-11-10T16:55:22.110Z	info	internal/resourcedetection.go:125	began detecting resource information	{"kind": "processor", "name": "resourcedetection/eks", "pipeline": "logs"}
2023-11-10T16:55:25.310Z	warn	eks/detector.go:97	Unable to get EKS cluster name	{"kind": "processor", "name": "resourcedetection/eks", "pipeline": "logs", "error": "EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestCanceled: request context canceled\ncaused by: context deadline exceeded"}
2023-11-10T16:55:25.310Z	info	internal/resourcedetection.go:139	detected resource information	{"kind": "processor", "name": "resourcedetection/eks", "pipeline": "logs", "resource": {"cloud.platform":"aws_eks","cloud.provider":"aws"}}
2023-11-10T16:55:25.310Z	info	adapter/receiver.go:45	Starting stanza receiver	{"kind": "receiver", "name": "filelog", "data_type": "logs"}
2023-11-10T16:55:25.310Z	info	service@v0.88.1-0.20231026220224-6405e152a2d9/service.go:169	Everything is ready. Begin running and processing data.
2023-11-10T16:55:25.511Z	info	fileconsumer/file.go:251	Started watching file	{"kind": "receiver", "name": "filelog", "data_type": "logs", "component": "fileconsumer", "path": "/var/log/pods/analytics-tests-rv_quantizer-alpha-0_551d1187-3e40-4179-a5fb-3b752da44ad6/quantizer/0.log"}
2023-11-10T16:55:28.811Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 5}
2023-11-10T16:55:32.611Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 2}
2023-11-10T16:55:41.810Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 3}
2023-11-10T16:55:44.610Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 2}
2023-11-10T16:55:50.211Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 2}
2023-11-10T16:56:00.211Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 6}
2023-11-10T16:56:01.210Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 4}
2023-11-10T16:56:02.811Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 10}
2023-11-10T16:56:04.011Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 6}
2023-11-10T16:56:05.011Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 4}
2023-11-10T16:56:06.713Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 100}
2023-11-10T16:56:06.811Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 14}
2023-11-10T16:56:08.611Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 1}
2023-11-10T16:56:08.810Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 3}
2023-11-10T16:56:11.411Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 3, "log records": 7}
2023-11-10T16:56:11.811Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 6}
2023-11-10T16:56:15.211Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 7}
2023-11-10T16:56:16.211Z	info	LogsExporter	{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 2, "log records": 4}
```
I agree the manual setting of `err = nil` should be improved. I think it would be fair to log each error in `getClusterName` at the warn level, and simply return an empty string. No harm is done by setting the k8s.cluster.name to be empty, so I that sounds like a fair solution to me.